### PR TITLE
feat: show metadata

### PR DESCRIPTION
**Pull request by dylancom**
#244

### CODE REVIEW

File: src/app/screens/LNURLPay.tsx [hunk 1]
```diff
@@ -130,15 +131,36 @@ function LNURLPay({ details, origin }: Props) {
...
 
+  function formattedMetadata() {
+    return JSON.parse(details.metadata)
+      .map(([type, content]: [string, string]) => {
+        if (type === "text/plain") {
+          return ["Description", content];
+        } else if (type === "text/long-desc") {
+          return ["Full Description", <p>{content}</p>];
+        } else if (["image/png;base64", "image/jpeg;base64"].includes(type)) {
+          return ["lnurl", <img src={content} alt="lnurl" />];
```
[the remark this answers is not in the document]
Comment (dylancom) [reply src/app/screens/LNURLPay.tsx:142]: I'm not aware of any base64 risks.
React always escapes unless you tell it not to.

File: src/app/screens/LNURLPay.tsx [hunk 2]
```diff
@@ -130,15 +131,36 @@ function LNURLPay({ details, origin }: Props) {
     }
   }
 
+  function formattedMetadata() {
+    return JSON.parse(details.metadata)
```
[the remark this answers is not in the document]
Comment (dylancom) [reply src/app/screens/LNURLPay.tsx:135]: Correct, will improve the func.

File: src/app/screens/LNURLPay.tsx [hunk 3]
```diff
@@ -130,15 +131,45 @@ function LNURLPay({ details, origin }: Props) {
...
+            return [
+              "lnurl",
+              <img src={`data:${type},${content}`} alt="lnurl" />,
+            ];
+          }
+          return undefined;
+        })
+        .filter(Boolean);
+    } catch (e) {
+      console.log(e.message);
```
Comment (bumi) [inline src/app/screens/LNURLPay.tsx:153]:
```suggestion
      console.log(e);
```

people don't see the console anyway and for use the whole error might be more useful to debug.

Comment (dylancom) [reply src/app/screens/LNURLPay.tsx]: Oke, to be more precise I refactored it to using console.error(e);

Comment (bumi) [reply src/app/screens/LNURLPay.tsx]: ah, good point. I must remember that for the other similar cases.

**Pull request merged**